File: Notes/app/src/main/java/com/serhii/apps/notes/ui/view_model/NotesViewModelFactory.java
package com.serhii.apps.notes.ui.view_model;

import android.app.Application;
import android.content.Context;

import androidx.annotation.NonNull;
import androidx.lifecycle.ViewModel;
import androidx.lifecycle.ViewModelProvider;

import com.serhii.core.log.Log;

/**
 *  View view factory for {@link com.serhii.apps.notes.ui.view_model.NotesViewModel} class
 */

public class NotesViewModelFactory extends ViewModelProvider.AndroidViewModelFactory {

    private static final String TAG = "NotesViewModelFactory";

    public NotesViewModelFactory(@NonNull Application application) {
        super(application);
    }

    @NonNull
    @Override
    public <T extends ViewModel> T create(@NonNull Class<T> modelClass) {
        Log.info(TAG, "create(), IN");
        T viewModel = super.create(modelClass);
        Log.info(TAG, "create(), created: " + viewModel);
        return viewModel;
    }

}

File: Notes/app/src/main/java/com/serhii/apps/notes/ui/view_model/NotesViewModelFactory.kt
/*
 * Copyright 2022. Happy coding ! :)
 * Author: Serhii Butryk
 */
package com.serhii.apps.notes.ui.view_model

import com.serhii.core.log.Log.Companion.info
import android.app.Application
import androidx.lifecycle.ViewModelProvider.AndroidViewModelFactory
import androidx.lifecycle.ViewModel
import com.serhii.apps.notes.ui.view_model.NotesViewModelFactory

/**
 * View view factory for [com.serhii.apps.notes.ui.view_model.NotesViewModel] class
 */
class NotesViewModelFactory(application: Application) : AndroidViewModelFactory(application) {

    override fun <T : ViewModel> create(modelClass: Class<T>): T {
        info(TAG, "create(), IN")
        val viewModel = super.create(modelClass)
        info(TAG, "create(), created: $viewModel")
        return viewModel
    }

    companion object {
        private const val TAG = "NotesViewModelFactory"
    }
}